src/components/lib/cos_kernel_api.c

```diff
@@ -200,7 +201,7 @@ __capid_captbl_check_expand(struct cos_compinfo *ci)
 
 	printd("__capid_captbl_check_expand->pre-captblactivate (%d)\n", CAPTBL_OP_CAPTBLACTIVATE);
 	/* captbl internal node allocated with the resource provider's captbls */
-	if (call_cap_op(meta->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, captblcap, meta->pgtbl_cap, kmem, 1)) {
+	if (call_cap_op(meta->captbl_cap, CAPTBL_OP_CAPTBLACTIVATE, captblcap, meta->local_mi.pgtbl_cap, kmem, 1)) {
```

Comment:
I think the current semantics of OP_CONS is fine.
It connects a subtree in **our own** cap table to a node within anther cap table which needs to be explicated provided.
**our own** cap table is always implicitly provided, unless we want to modify it. 
Whenever we want to modify a cap table including our own cap table, we have to provide a cap to it.
But we read some caps in **our own** cap table, we do not need to provide a captbl_cap to **our own** cap table.